controller/internal/enforcer/nfqdatapath/datapath.go

```diff
@@ -328,7 +389,8 @@ func (d *Datapath) Enforce(contextID string, puInfo *policy.PUInfo) error {
 	if err != nil {
 		return fmt.Errorf("error creating new pu: %s", err)
 	}
-
+	// this context pointer is about to get lost. reclaims its counters
+	d.puCountersChannel <- pu
```

Comment:
Weird things can happen here: 
1. If the channel blocks, we will get blocked here (and we have see this several times).

2. Can a PU be multiple times in the channel? I don't see why not. What will the effect of this be?